### DISCUSSION TITLE
How to use Sowfa.utilities in Floris?

### DISCUSSION THREAD

**Discussion by zevenlordsa** [NatLabRockies/floris#688] [Q&A] [answered]
I am trying to use the wind field data from Sowfa for calculation in Floris, but now I don't know much about the use of sowfa.utilities script. Could you help me explain it? Some simple examples will be great. Thank you very much!

**Comment by paulf81** [comment, marked as answer]
@zevenlordsa, thank you for your question.  To be honest this module is now probably a little outdated.  A lot of our internal research has moved to using AMR-wind for LES simulations (https://github.com/Exawind/amr-wind) and because of this use of this module is not tracking updates to SOWFA.  That said, a way to understand the expected usage is if you check the __init__ function:

```
    def __init__(
        self,
        case_folder,
        flow_data_sub_path="array_mean/array.mean0D_UAvg.vtk",
        setup_sub_path="setUp",
        turbine_array_sub_path="constant/turbineArrayProperties",
        turbine_sub_path="constant/turbineProperties",
        controlDict_sub_path="system/controlDict",
        turbine_output_sub_path="turbineOutput/20000",
        assumed_settling_time=None,
    ):
```

You can see it the files it expects to exist within the SOWFA simulation case folder you specify by default (you have the option to specify others).  The VTK output file of flow data would come from the output averaged UAvg over the volume at some time in the simulation.  The turbine output (note we are expecting an actuator-disk type simulation).

But in general I think this module might need to be deprecated since it's not being updated (@rafmudaf flagging your opinion?).  We are however working on a separate repository for interacting with AMR-wind outputs with FLORIS.  But I am sorry my response is not more helpful

**Reply by zevenlordsa** [reply]
Thank you so much! Really appreciate.Since this module is not updated, I will try something else. : )
